Replace snowflake test password auth with key pair auth

rileythomp opened the pull request:
### Description

Replaces snowflake password auth with key pair auth for CI tests

### How to verify

All tests should pass

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

rileythomp committed:
replace password with key pair auth

rileythomp committed:
re enable snowflake tests

rileythomp committed:
add private key env var

rileythomp committed:
Merge branch 'master' into update-snowflake-test-auth

rileythomp committed:
Merge branch 'master' of https://github.com/metabase/metabase into update-snowflake-test-auth

rileythomp committed:
Merge branch 'master' into update-snowflake-test-auth

rileythomp committed:
update private key details for temp dbs

rileythomp committed:
Merge branch 'update-snowflake-test-auth' of https://github.com/metabase/metabase into update-snowflake-test-auth

rileythomp committed:
small changes

rileythomp committed:
fix describe database test

rileythomp committed:
fix multi method params

rileythomp committed:
update test

rileythomp committed:
update test

rileythomp committed:
try fix

dpsutton committed:
check CI

i noticed that this tests pass if we don't put a connection-uri in
there. I'm not sure what the original intent was, but the following
works just fine:

```
{:url "jdbc:snowflake://<account>.snowflakecomputing.com/?schema=INFORMATION_SCHEMA",
 :properties {"use-password" "false",
              "connection-uri" "jdbc:snowflake://<account>.snowflakecomputing.com/?user=<user>CI&private_key_file=%2Fprivate%2Fvar%2Ffolders%2F1d%2F3ns5s1gs7xjgb09bh1yb6wpc0000gn%2FT%2Fmetabase-secret_8808579641027884666.tmp&schema=INFORMATION_SCHEMA",
              "client_session_keep_alive" "true",
              "timezone" "UTC",
              "warehouse" "<warehouse>",
              "user" "<user>",
              "ssl" "true",
              "private_key_file" "/var/folders/1d/3ns5s1gs7xjgb09bh1yb6wpc0000gn/T/metabase-secret_8808579641027884666.tmp",
              "week_start" "7",
              "application" "Metabase_Metabase",
              "account" "<account>",
              "db" "\"<db>\"",
              "time_output_format" "HH24:MI:SS.FF3",
              "client_metadata_request_use_connection_ctx" "true"}}
```

When it goes through clojure.java.jdbc. Somehow we are getting a
connection string that doesn't play nice. I _think_ it may just be an
issue with snowflake but i'm not positive.

dpsutton committed:
remove parallel test markers

dpsutton committed:
remove parallel marker

when running tests locally with these removed:

```clojure
(run/find-and-run-tests-repl {:only 'metabase.driver.snowflake-test})
...
Ran 49 tests in 116.525 seconds
502 assertions, 1 failure, 0 errors.
{:test 49, :pass 501, :fail 1, :error 0, :type :summary, :duration 116525.006833, :single-threaded 32, :parallel 17}
```
yields only one expected error. Previously using regular stuff also
worked

```clojure
snowflake-test=> (run-tests)

Testing metabase.driver.snowflake-test

Ran 49 tests containing 502 assertions.
1 failures, 0 errors.
{:test 49, :pass 501, :fail 1, :error 0, :type :summary}
```

(again the single test that failed). but using the CLI tool i would get
errors when running them in parallel.

dpsutton committed:
fixup stupid snowflake

1. i don't know why we are running into multithreading issues now. But
i've removed those tags and the issues are gone.
2.

```clojure
snowflake-test=> (let [spec {:connection-uri "jdbc:snowflake://account.snowflakecomputing.com/?schema=INFORMATION_SCHEMA&user=<user>&password=<password>&db=%22v5_airports%22"}]
                   [(jdbc/query spec ["select CURRENT_DATABASE() db"])
                    (jdbc/query spec ["select CURRENT_SCHEMA() db"])])
 [({:db "v5_airports"}) ({:db "INFORMATION_SCHEMA"})]
snowflake-test=> (let [spec {:connection-uri "jdbc:snowflake://account.snowflakecomputing.com/?schema=INFORMATION_SCHEMA&user=<user>&password=<password>&db=v5_airports"}]
                   [(jdbc/query spec ["select CURRENT_DATABASE() db"])
                    (jdbc/query spec ["select CURRENT_SCHEMA() db"])])
[({:db nil}) ({:db nil})]
```

In the above, note that putting the db in the connection string _and
url-encoding it_ (it's already quoted) is necessary.

dpsutton committed:
conditionally add db

in tests there's not a database so just be smart about it:

```clojure
{:warehouse "<warehouse>",
 :additional-options nil,
 :timezone "UTC",
 :db nil,
 :account "<account>",
 :use-password false,
 :time_output_format "HH24:MI:SS.FF3",
 :user "<user>",
 :subname "//<account>.snowflakecomputing.com/"}
```

rileythomp committed:
Merge branch 'master' into update-snowflake-test-auth

bshepherdson commented on modules/drivers/snowflake/test/metabase/driver/snowflake_test.clj:
```diff
@@ -249,13 +249,17 @@
          (is (sql-jdbc.sync/have-select-privilege? :snowflake conn "PUBLIC" "venues")))))))
 
-(deftest ^:parallel can-set-schema-in-additional-options
+(deftest can-set-schema-in-additional-options
```
And through the rest of the file too. (I'm betting this was a bad automatic merge, though?)

bshepherdson reviewed (changes requested):
Withholding the LGTM bit for now because of that worrying comment.

rileythomp committed:
address pr comments

dpsutton commented on modules/drivers/snowflake/src/metabase/driver/snowflake.clj:
```diff
@@ -128,6 +130,7 @@
       (-> details
           (driver-api/clean-secret-properties-from-details :snowflake)
+          ;; don't ship like this, just checking CI
           (handle-conn-uri user account private-key-file)
```
@bshepherdson good call out. I put that comment in when i _commented out_ the `handle-conn-uri`. It was saying not to ship a transient state that is no longer present. You are **100%** correct to require changes on this though. Thank you

dpsutton committed:
remove outdated comment

pretty freaky comment eh? I added this when i commented out the
`handle-conn-uri` call underneath it and it caused tests to pass. That
was helpful in identifying the mechanism, but not a good solution. Put a
comment here so that that was not a final state, forgot to remove
comment when we got to a proper final state and `handle-conn-uri` was
fixed.
